app/controllers/users/sessions_controller.rb

```diff
@@ -162,7 +162,7 @@ def alive?
     end
 
     def track_authentication_attempt(email)
-      user = User.find_with_email(email) || AnonymousUser.new
+      user = current_user || User.find_with_email(email) || AnonymousUser.new
```

Comment:
I'm trying to understand how a user would already be signed in when reaching this code. Are users routed through this controller for an already-logged-in session? Do we even want to be tracking the authentication attempt in that scenario?

One concern I have with this change is that the method's argument would give me the impression that the authentication attempt would be associated with the email address passed to the method, when in-fact we may disregard it if there's a current user already.

I wonder if we could remove the argument and reference the parameters directly in the fallback?

```rb
def track_authentication_attempt
  user = current_user || User.find_with_email(auth_params[:email]) || AnonymousUser.new
```